activity.php

```diff
@@ -19,37 +22,31 @@ public function printData(){
         }
 
         // define contracts for basic fetch and mutate functionality
-        abstract public function getActivities() : iterable;
-        abstract public function getActivity($key);
-        abstract public function addActivity($key, $value);
-        abstract public function deleteActivity($key): iterable;
+        abstract public function get() : Ds\Sequence;
+        abstract public function add($key, $value);
+        abstract public function delete($key) : bool;
     }
 
     class ActivityImpl extends Activity {
-        public function getActivities(): iterable {
-            return $this->data;
+        public function get() : Ds\Sequence {
```

Comment:
Accept `key` and return value in get. Then create an `all` method to return all entries